readme.md

```diff
@@ -15,834 +14,1113 @@
+// When only evaluating that an array has length,
+// instead of this:
+if ( array.length > 0 ) ...
 
-    var neg = -2.5;
+// ...evaluate truthiness, like this:
+if ( array.length ) ...
 
-    parseInt( neg, 10 );
+// When only evaluating that an array is empty,
+// instead of this:
+if ( array.length === 0 ) ...
 
-    // Is the same as...
+// ...evaluate truthiness, like this:
+if ( !array.length ) ...
+```
 
-    ~~neg;
+### 2.4.2. Boolean string evaluation
+```javascript
+// When only evaluating that a string is not empty,
+// instead of this:
+if ( string !== '' ) ...
 
-    neg >> 0;
+// ...evaluate truthiness, like this:
+if ( string ) ...
 
-    // All result in -2.
-    // However...
+// When only evaluating that a string _is_ empty,
+// instead of this:
+if ( string === '' ) ...
 
-    neg >>> 0;
+// ...evaluate falsyness, like this:
+if ( !string ) ...
+```
 
-    // Will result in 4294967294.
+### 2.4.3. Boolean evaluation
 
-    ```
+```javascript
+// When only evaluating that a reference is true, instead of this:
+if ( foo === true ) ...
 
+// ...evaluate like you mean it, take advantage of built in capabilities:
+if ( foo ) ...
 
-4. <a name="cond">Conditional Evaluation</a>
+// When evaluating that a reference is false, instead of this:
+if ( foo === false ) ...
 
-    ```javascript
+// ...use negation to coerce a true evaluation
+if ( !foo ) ...
 
-    // 4.1.1
-    // When only evaluating that an array has length,
-    // instead of this:
-    if ( array.length > 0 ) ...
+// ...Be careful, this will also match: 0, '', null, undefined, NaN. If you MUST test for a boolean false, then use:
+if ( foo === false ) ...
+```
 
-    // ...evaluate truthiness, like this:
-    if ( array.length ) ...
+### 2.4.4. Boolean ref evaluation
 
+```javascript
+// When only evaluating a ref that might be null or undefined,
+// but NOT false, '' or 0, instead of this:
+if ( foo === null || foo === undefined ) ...
 
-    // 4.1.2
-    // When only evaluating that an array is empty,
-    // instead of this:
-    if ( array.length === 0 ) ...
+// ...take advantage of == type coercion, like this:
+if ( foo == null ) ...
 
-    // ...evaluate truthiness, like this:
-    if ( !array.length ) ...
+// Remember, using == will match a `null` to BOTH `null` and `undefined` but not `false`, '' or 0
+null == undefined
+```
 
+ALWAYS evaluate for the best, most accurate result - the above is a guideline, not a dogma.
 
-    // 4.1.3
-    // When only evaluating that a string is not empty,
-    // instead of this:
-    if ( string !== '' ) ...
+### 2.4.5. Loose equality coersion 
+Use loose equality operator to simplify your code when you want to leverage type coercion. Remember that strict equality comparator `===` is checking if both the type and the value you are comparing are the same. In a constract, loose equality operator `==` will try to do type coersion which may be useful in some cases when comparing values where types are less revelant than values.
```

Comment:
I'd add a reference to a cheatsheet here, like [this one](https://dorey.github.io/JavaScript-Equality-Table/).

Reply:
I'm a bit skeptical about adding 3rd party references which are not official docs :thinking: Such links like to go out of date or provide incorrect information. Can you share some official docs instead?